src/services/json-api-datastore.service.ts

```diff
@@ -90,15 +92,23 @@ export class JsonApiDatastore {
         } else {
             httpCall = this.http.post(url, body, options);
         }
-        return httpCall
+        const modelObservable = httpCall
             .map(res => res.status === 201 ? this.extractRecordData(res, modelType, model) : model)
-            .catch(error => {
-                console.error(error);
-                return Observable.of(model);
-            })
-            .map(res => this.resetMetadataAttributes(res, attributesMetadata, modelType))
-            .map(res => this.updateRelationships(res, relationships))
-            .catch((res) => this.handleError(res));
+            .catch((res) => {
+                if (res == null) {
+                    return Observable.of(model);
+                }
+                return this.handleError(res)
+            });
+
+        return Observable.merge(
+            modelObservable
+                .filter(res => res.errors),
```

Comment:
where does the errors property come from. Isn´t the type of modelObservable `Observable<T>`?